### PR TITLE
Remote books frontend integration

### PR DESCRIPTION
**Pull request by caseyszilagyi**
## Feature Description/Implementation
- New fields on book list
- View differing intrinsic fields on book detail view
- Allow user to import editable intrinsic fields on book modify page (page count, dimensions, cover art)
- Allow user to import editable intrinsic fields on book add page

## Everything Else
- Will have to change implementation because of refactor
- I can't think of a clean way to disable the import image button if our image is the same as the subsidiary. This is because the URLs will always be different. I suppose I could compare the bytes of the image, but this seems excessive.
- Our subsidiary has lower resolution images, so we may need to change something to make it look nicer on our book detail page

### CODE REVIEW

File: frontend/src/components/datatable/DeleteColumn.tsx [hunk 1]
```diff
@@ -21,7 +22,7 @@ export default function DeleteColumn<T>(props: DeleteColumnProps<T>) {
   return (
     <Column
       body={rowDeleteButton}
-      header={"Delete"}
+      header={props.hideHeader ? "" : "Delete"}
```
Comment (nex-pixel) [inline frontend/src/components/datatable/DeleteColumn.tsx:25]: Why are we hiding header like this?

File: frontend/src/components/text/TextWrapperNullableNumberEditor.tsx [hunk 2]
```diff
@@ -21,14 +22,16 @@ export function TextWrapperNullableNumberEditor(
...
         <div className={props.valueClassName ?? ""}>
-          {NullableNumberEditor(
-            props.value,
-            (newValue: number | undefined) =>
-              props.onValueChange(newValue ?? props.defaultValue),
-            "w-4",
-            props.disabled,
-            props.min
-          )}
+          <p className="vertical-align-middle">
```
Comment (nex-pixel) [inline frontend/src/components/text/TextWrapperNullableNumberEditor.tsx:25]: why is this changed to vertical-align-middle? what does it do?

File: frontend/src/pages/books/BookAdd.tsx [hunk 3]
```diff
@@ -150,71 +153,181 @@ export default function BookAdd() {
...
-            setBooks((draft) => {
-              const book = findById(draft, rowData.id)!;
-              book.pageCount = newValue;
-            });
-          },
-          "integerNumbeBookAdd"
-        ),
+      customBody: (rowData: BookWithDBTag) => (
+        <>
+          <div>
```
Comment (nex-pixel) [inline frontend/src/pages/books/BookAdd.tsx:158]: Maybe it you should extract these components during refactorization.

File: frontend/src/pages/books/BookAdd.tsx [hunk 4]
```diff
@@ -150,71 +153,181 @@ export default function BookAdd() {
...
-          "decimalNumberBookAdd",
-          false,
-          0.01
-        ),
+      customBody: (rowData: BookWithDBTag) => (
+        <>
+          <div>
+            {NullableNumberEditor(
+              rowData.width,
+              (newValue) => {
```
Comment (nex-pixel) [inline frontend/src/pages/books/BookAdd.tsx:205]: same here.

File: frontend/src/pages/books/BookDetail.tsx [hunk 5]
```diff
@@ -565,44 +639,74 @@ export default function BookDetail() {
           <div className="flex col-12 justify-content-start p-1">
             <div className="flex p-0">
               <TextLabel label="Title:" />
-              <p className="p-component p-text-secondary text-900 text-3xl text-center my-0">
+              <p className="p-component p-text-secondary text-900 text-m text-center my-0">
                 {title}
+                <label className="text-blue-800">
+                  {remoteBook && remoteBook.title != title
+                    ? `\xa0 (Remote: ${remoteBook.title})`
```
Comment (nex-pixel) [inline frontend/src/pages/books/BookDetail.tsx:646]: Can you leave a comment as to what the \xa0 means and this thing does.

File: frontend/src/pages/books/BookDetail.tsx [hunk 6]
```diff
@@ -565,44 +639,74 @@ export default function BookDetail() {
...
             </div>
           </div>
           <div className="flex col-12 justify-content-start p-1">
             <div className="flex p-0">
               <TextLabel label="Author(s):" />
-              <p className="p-component p-text-secondary text-900 text-2xl text-center m-0">
+              <p className="p-component p-text-secondary text-900 text-m text-center m-0">
                 {authors}
+                <label className="text-blue-800">
+                  {remoteBook && remoteBook.authors != authors
```
Comment (nex-pixel) [inline frontend/src/pages/books/BookDetail.tsx:658]: same for others/

**Review by nex-pixel — CHANGES_REQUESTED** [review]

File: frontend/src/apis/books/BooksConversions.tsx [hunk 7]
```diff
@@ -164,5 +168,32 @@ export function APIToInternalBookConversionWithDB(
...
+    isbn10: book.isbn10,
+    publisher: book.publisher,
+    publishedYear: book.publicationYear,
+    pageCount: book.pageCount,
+    width: book.width,
+    height: book.height,
+    thickness: book.thickness,
+    retailPrice: book.retailPrice,
+    stock: book.inventoryCount,
+    // Temporary workaround because subsidiary gives http instead of https
```
[the remark this answers is not in the document]
Comment (caseyszilagyi) [reply frontend/src/apis/books/BooksConversions.tsx:193]: I'll change the comment but going to leave the code, I think it's a good safeguard and doesn't do any harm

File: frontend/src/components/datatable/DeleteColumn.tsx (hunk 1 above)
Comment (caseyszilagyi) [reply frontend/src/components/datatable/DeleteColumn.tsx:25]: I was experimenting with saving space, but I don't think it does much so I will remove it

File: frontend/src/components/text/TextWrapperNullableNumberEditor.tsx (hunk 2 above)
Comment (caseyszilagyi) [reply frontend/src/components/text/TextWrapperNullableNumberEditor.tsx:25]: I had to make the input fields smaller so that I could fit the import field buttons, but making them smaller made them off-centered with the label. I added the vertical-align-middle to fix this

File: frontend/src/pages/books/BookAdd.tsx (hunk 3 above)
Comment (caseyszilagyi) [reply frontend/src/pages/books/BookAdd.tsx:158]: Noted, I will do this when I do the refactor for updating the endpoint

File: frontend/src/pages/books/BookDetail.tsx (hunk 5 above)
Comment (caseyszilagyi) [reply frontend/src/pages/books/BookDetail.tsx:647]: \xa0 adds a space, but I realized I can just use padding so I removed it

**Review by nex-pixel — APPROVED** [review]
